packages/x-components/src/x-modules/tagging/service/pdp-add-to-cart.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Store } from 'vuex';
+import { Result } from '@empathyco/x-types';
+import { BrowserStorageService, StorageService } from '@empathyco/x-storage-service';
+import { Logger, logger } from '@empathyco/x-logger';
+import { RootXStoreState } from '../../../store/index';
+import { XPlugin } from '../../../plugins/index';
+import { PDPAddToCartService } from './types';
+
+/**
+ * Default implementation for the {@link PDPAddToCartService}.
+ *
+ * @public
+ */
+export class DefaultPDPAddToCartService implements PDPAddToCartService {
+  protected logger: Logger;
+  /**
+   * Session id key to use as key in the storage.
+   *
+   * @public
+   */
+  public static readonly RESULT_CLICKED_ID_KEY = 'add-to-cart';
+
+  /**
+   * Global instance of the {@link PDPAddToCartService}.
+   */
+  public static instance: PDPAddToCartService = new DefaultPDPAddToCartService();
+
+  public constructor(
+    protected localStorageService: StorageService = new BrowserStorageService(localStorage, 'x'),
+    protected sessionStorageService: StorageService = new BrowserStorageService(sessionStorage, 'x')
+  ) {
+    this.logger = logger.child(`[PDPAddToCartService]`);
+  }
+
+  protected get store(): Store<RootXStoreState> {
+    return XPlugin.store;
+  }
+
+  /**
+   * Stores in the local storage the information from the Result clicked by the user
+   * in order to be able to track later on.
+   *
+   * @param result - The result to store.
+   */
+  storeResultClicked(result: Result): void {
+    const { clickedResultStorageKey, clickedResultStorageTTLMs } =
+      this.store.state.x.tagging.config;
+
+    let key = result[clickedResultStorageKey as keyof Result] as string;
+
+    if (clickedResultStorageKey === 'url') {
+      key = key.replace(/\s|\+/g, '%20');
+    }
+
+    const storageKey = `${DefaultPDPAddToCartService.RESULT_CLICKED_ID_KEY}-${key}`;
+    this.localStorageService.setItem(storageKey, result, clickedResultStorageTTLMs as number);
+  }
+
+  /**
+   * Checks if the local storage contains a result information for the given id and moves
+   * the result info from the local storage to the session storage.
+   *
+   * @param id - The id of the result to move to the session storage.
+   */
+  moveToSessionStorage(id: string): void {
+    let clickedResultStorageKeyId: string;
+
+    if (id === 'url') {
+      const url = window.location.href.replace(/\s|\+/g, '%20');
+      clickedResultStorageKeyId = `${DefaultPDPAddToCartService.RESULT_CLICKED_ID_KEY}-${url}`;
+    } else {
+      clickedResultStorageKeyId = `${DefaultPDPAddToCartService.RESULT_CLICKED_ID_KEY}-${id}`;
+    }
+
+    const result = this.localStorageService.getItem(clickedResultStorageKeyId);
```

Comment:
the `localStorageService.removeItem` already returns the item or null. So you can do the get and the remove in one single step.